include/cantera/kinetics/Arrhenius.h

```diff
@@ -195,6 +198,102 @@ class ArrheniusRate final : public ArrheniusBase
 };
 
 
+//! Two temperature plasma reaction rate type depends on both
+//! gas temperature and electron temperature.
+/*!
+ * The form of the two temperature plasma reaction rate coefficient is similar to an
+ * Arrhenius reaction rate coefficient. The temperature exponent (b) is applied to
+ * the electron temperature instead. In addition, the exponential term with
+ * activation energy for electron is included.
+ *
+ *   \f[
+ *        k_f =  A T_e^b \exp (-E_a/RT) \exp (-E_{a,e}/RT_e)
+ *   \f]
+ *
+ * Ref.: Kossyi, I. A., Kostinsky, A. Y., Matveyev, A. A., & Silakov, V. P. (1992).
+ * Kinetic scheme of the non-equilibrium discharge in nitrogen-oxygen mixtures.
+ * Plasma Sources Science and Technology, 1(3), 207.
+ * doi: 10.1088/0963-0252/1/3/011
+ * @ingroup arrheniusGroup
+ */
+class TwoTempPlasmaRate final : public ArrheniusBase
+{
+public:
+    TwoTempPlasmaRate();
+
+    //! Constructor.
+    /*!
+     *  @param A  Pre-exponential factor. The unit system is (kmol, m, s); actual units
+     *      depend on the reaction order and the dimensionality (surface or bulk).
+     *  @param b  Temperature exponent (non-dimensional)
+     *  @param Ea  Activation energy in energy units [J/kmol]
+     *  @param EE  Activation electron energy in energy units [J/kmol]
+     */
+    TwoTempPlasmaRate(double A, double b, double Ea, double EE);
+
+    unique_ptr<MultiRateBase> newMultiRate() const override {
+        return unique_ptr<MultiRateBase>(
+            new MultiBulkRate<TwoTempPlasmaRate, TwoTempPlasmaData>);
+    }
+
+    //! Constructor based on AnyMap content
+    TwoTempPlasmaRate(const AnyMap& node, const UnitStack& rate_units={})
+        : TwoTempPlasmaRate()
+    {
+        setParameters(node, rate_units);
+    }
+
+    //! Identifier of reaction rate type
+    virtual const std::string type() const override {
+        return "two-temperature-plasma";
+    }
+
+    virtual void setRateParameters(const AnyValue& rate,
+                                   const UnitSystem& units,
+                                   const UnitStack& rate_units) override;
+
+    //! Perform object setup based on AnyMap node information
+    /*!
+     *  @param node  AnyMap containing rate information
+     *  @param rate_units  Unit definitions specific to rate information
+     */
+    virtual void setParameters(const AnyMap& node, const UnitStack& rate_units) override;
+
+    virtual void getParameters(AnyMap& node) const override;
+
+    double evalFromStruct(const TwoTempPlasmaData& shared_data) {
+        return m_A * std::exp(m_b * shared_data.logTe -
```

Comment:
@BangShiuh this is not equivalent to the form given in the docstrings above, isn't it? 
```
k_f = A T_e^b \exp (-E_a/RT) \exp (-E_{a,e}/RT_e)
```
which is the form I would expect in 2-T plasma

Reply:
TwoTempPlasma is no longer in Arrhenius. The docstrings should be update-to-date in TwoTempPlasmaRate.h.

Reply:
I'm don't think the currently implemented form would work for (at least) our plasma models. 

The currently implemented form is (1) :  
```
k_f =  A T_e^b \exp (-E_{a,g}/RT) \exp (E_{a,e} (T_e - T)/(R T T_e))
```

For T_e >> T  which is typical in the discharge phase, it reduces to (2) : 
```
k_f = A T_e^b \exp (E_{a,e}-E_{a,g}/RT)
``` 

This doesn't let you model Ionization and Excitation reactions. 

In Kossyi's paper, these reactions are modeled as a function of the reduced electric field
For instance nitrogen ionization below : 

![image](https://user-images.githubusercontent.com/16088743/189422205-1c5cc8e8-d87a-4095-b13f-cd383d39f59d.png)

In a 2-T model you could write them as a function of electron temperature. E/N is proportional to Te in the drift-diffusion approximation. You could convert these to an Arrhenius form, where the temperature is electron temperature :  

```
k_f = A T_e^b \exp (-E_a/RT) \exp (-E_{a,e}/RT_e)
```

However, the form currently implemented in Cantera won't help. 

It seems to be good to model electron attachment and detachment processes, as used by Kosssyi, but unfortunately these are not the main mechanisms in (at least) our plasma-assisted-combustion models. 

---

How hard would it be to add the Electron-temperature Arrhenius form back ? 
I'm really not familiar with C so cannot help unfortunately.

Reply:
$k_f =  A T_e^b \exp (-E_{a,g}/RT) \exp (E_{a,e} (T_e - T)/(R T T_e))$
is the same as
$k_f =  A T_e^b \exp (-E_{a,g}/RT) \exp (E_{a,e}/(R T)) \exp(-E_{a,e}/(R T_e))$
You need to cancel $\exp (E_{a,e}/(R T))$ by setting E_{a,g} = E_{a,e}.

Reply:
I am working on ElectronCollisionPlasmaRate which will use the cross-section data and electron energy distribution function to calculate the reaction rates. Thank you for your interest in this feature! We would like to see more people contribute to this!

Reply:
You're right. It might be hard to read and slightly slower to compute, but it works so that's enough. 

Your end goal is challenging but very helpful. How do you solve the electron energy distribution ? Bolsig / Bolos ?

Reply:
I will write an electron Boltzmann equation solver similar to BOLOS in C++.

Reply:
I'd use this right-away 👏  🚀

Reply:
It might be worth an update to the documentation to note this use case of setting $E_{a,g} = E_{a,e}$ to support reactions with the rate expression desired here, since that transformation of the rate expression might not occur to users.